### PR TITLE
feat: implement `assert_any_suggestion_result` as in #950

### LINKED ISSUE #950: feat(core): Add an `assert_any_suggestion_result` test function
It would be nice to have a testing assertion along with `assert_suggestion_result()` and `assert_nth_suggestion_result()` that passes of the test no matter which suggestion appears in the result.

I looked at implementing this briefly a week or two ago and decided I didn't know enough about the architecture to do it efficiently.

### PR DESCRIPTION
**Pull request by hippietrail**
# Issues 
Closes #950 

# Description

Implements `assert_any_suggestion_result` so you don't have to include the exact suggestion number that you expect to match.

# How Has This Been Tested?

I added some unit tests using one of the linters that suggests more than one replacement, `possess_your`.
In doing so I noticed it returned one option with "an" but no matching one with "a", so I added that too.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have added tests to cover my changes

### CODE REVIEW

**Review by elijah-potter — CHANGES_REQUESTED** [review]
I have some minor edits, but this is very close. Thank you for tackling the issue.

File: harper-core/src/linting/possessive_your.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ impl PatternLinter for PossessiveYour {
             lint_kind: LintKind::WordChoice,
             suggestions: vec![
                 Suggestion::replace_with_match_case("your".chars().collect(), orig_chars),
+                Suggestion::replace_with_match_case("you're a".chars().collect(), orig_chars),
```
[the remark this answers is not in the document]
Comment (hippietrail) [reply harper-core/src/linting/possessive_your.rs:41]: I think spotting edge cases like that is one of the things I'm good at.

File: harper-core/src/linting/mod.rs [hunk 2]
```diff
@@ -239,4 +239,61 @@ mod tests {
...
+        let lints = linter.lint(&test);
+
+        if lints.iter().any(|lint| {
+            lint.suggestions.iter().any(|suggestion| {
+                let mut text_chars = text.chars().collect::<Vec<_>>();
+                suggestion.apply(lint.span, &mut text_chars);
+                text_chars.iter().collect::<String>() == expected_result
+            })
+        }) {
+            return;
```
[the remark this answers is not in the document]
Comment (hippietrail) [reply harper-core/src/linting/mod.rs:265]: Not so minor after all! Looking into this and comparing `_nth_` and `_any_` made me realize mine does not handle multiple corrections. This will take some more effort to get right.

File: harper-core/src/linting/possessive_your.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (elijah-potter) [reply harper-core/src/linting/possessive_your.rs:43]: I'm in very solid agreement.

**Review by elijah-potter — APPROVED** [review]
Great work! I'll start using this in some tests soon. I'll let you know if I run into any issues. Thanks!
